nimbus-jose-jwt_aws-kms-extension/src/main/java/com/nimbusds/jose/aws/kms/crypto/impl/KmsAsymmetricSigningCryptoProvider.java

```diff
@@ -0,0 +1,126 @@
+     */
+    @NonNull
+    @Getter(AccessLevel.PROTECTED)
+    private final MessageType messageType;
+
+    public static final Map<JWSAlgorithm, String> JWS_ALGORITHM_TO_MESSAGE_DIGEST_ALGORITHM =
+            ImmutableMap.<JWSAlgorithm, String>builder()
+                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.RSASSA_PKCS1_V1_5_SHA_256.toString()),
+                            MessageDigestAlgorithms.SHA_256)
+                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.RSASSA_PKCS1_V1_5_SHA_384.toString()),
+                            MessageDigestAlgorithms.SHA_384)
+                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.RSASSA_PKCS1_V1_5_SHA_512.toString()),
+                            MessageDigestAlgorithms.SHA_512)
+                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.RSASSA_PSS_SHA_256.toString()),
+                            MessageDigestAlgorithms.SHA_256)
+                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.RSASSA_PSS_SHA_384.toString()),
+                            MessageDigestAlgorithms.SHA_384)
+                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.RSASSA_PSS_SHA_512.toString()),
+                            MessageDigestAlgorithms.SHA_512)
+                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.ECDSA_SHA_256.toString()),
+                            MessageDigestAlgorithms.SHA_256)
+                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.ECDSA_SHA_384.toString()),
+                            MessageDigestAlgorithms.SHA_384)
+                    .put(JWSAlgorithm.parse(SigningAlgorithmSpec.ECDSA_SHA_512.toString()),
+                            MessageDigestAlgorithms.SHA_512)
+                    .build();
+
+    /**
+     * The supported JWS algorithms (alg) by the RSA-SSA provider class.
+     * <p>
+     * Note: We are using KMS prescribed algorithm names here.
+     * Ref: https://docs.aws.amazon.com/kms/latest/developerguide/symm-asymm-choose.html#key-spec-rsa-sign
+     */
+    public static final Set<JWSAlgorithm> SUPPORTED_ALGORITHMS = JWS_ALGORITHM_TO_MESSAGE_DIGEST_ALGORITHM.keySet();
+
+    protected KmsAsymmetricSigningCryptoProvider(
+            @NonNull final AWSKMS kms, @NonNull final String privateKeyId, @NonNull final MessageType messageType) {
+        super(SUPPORTED_ALGORITHMS);
+        this.kms = kms;
+        this.privateKeyId = privateKeyId;
+        this.messageType = messageType;
+    }
+
+    protected ByteBuffer getMessage(final JWSHeader header, final byte[] signingInput) throws JOSEException {
+        final var alg = header.getAlgorithm();
+        var message = signingInput;
+
+        String messageDigestAlgorithm = Optional.ofNullable(JWS_ALGORITHM_TO_MESSAGE_DIGEST_ALGORITHM.get(alg))
+                .orElseThrow(() -> new JOSEException(
+                        String.format("No digest algorithm exist for JWE algorithm %s in map: %s",
```

Comment:
This should rather be JWS algorithm. (JWE mentioned)

Reply:
Fixed.